devito/ir/support/space.py

```diff
@@ -954,10 +970,43 @@ def prefix(self, key):
         try:
             i = self.project(key)[-1]
         except IndexError:
-            return None
+            return null_ispace
 
         return self[:self.index(i.dim) + 1]
 
+    def split(self, d):
+        """
+        Split the IterationSpace into two IterationSpaces, the first
+        containing all Intervals up to and including the one defined
+        over Dimension `d`, and the second containing all Intervals starting
+        from the one defined over Dimension `d`.
+        """
+        if isinstance(d, IterationInterval):
+            d = d.dim
+
+        if d is None:
+            return IterationSpace([]), self
```

Comment:
why not self first?

Reply:
None often represents the degenerate iteration space, that is an Iteration space with no dimensions in it (for equations that appear outside of any loops)